### PR TITLE
Add `crictl [create,runp,run,pull] jsonschema` subcommands

### PR DESCRIPTION
**Pull request by saschagrunert**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:
The `jsonschema` / `js` subcommands will print the pod and/or container config JSON schemas for end users to understand where the data is coming from.

#### Which issue(s) this PR fixes:

Refers to https://github.com/kubernetes/website/issues/40360
#### Special notes for your reviewer:
None
#### Does this PR introduce a user-facing change?

```release-note
Added `crictl [create,runp,run,pull] jsonschema` subcommands to display the pod and/or container config JSON schema.
```

### CODE REVIEW

**Comment by saschagrunert** [comment]
@kubernetes-sigs/cri-tools-maintainers PTAL

File: cmd/crictl/util.go [hunk 1]
```diff
@@ -170,6 +171,16 @@ func loadContainerConfig(path string) (*pb.ContainerConfig, error) {
 	return &config, nil
 }
 
+func printJSONSchema(typ any) error {
```
[the remark this answers is not in the document]
Comment (saschagrunert) [reply cmd/crictl/util.go:174]: `type` is a reserved keyword in golang. I can change it to something else. :)

**Comment by saschagrunert** [comment]
@kubernetes-sigs/cri-tools-maintainers PTAL

File: cmd/crictl/container.go [hunk 2]
```diff
@@ -169,12 +169,50 @@ var runPullFlags = []cli.Flag{
...
+		},
+	},
+	UseShortOptionHandling: true,
+	Action: func(c *cli.Context) error {
+		if !c.IsSet("pod") && !c.IsSet("container") {
+			return cli.ShowSubcommandHelp(c)
+		}
+
+		if c.IsSet("pod") {
+			if err := printJSONSchema(&pb.PodSandboxConfig{}); err != nil {
```
Comment (SergeyKanzhelev) [inline cmd/crictl/container.go:195]: this will print the schema of the version that crictl was compiled with, not the runtime version, correct?

Comment (saschagrunert) [reply cmd/crictl/container.go:195]: Yes, that's true.

File: cmd/crictl/image.go [hunk 3]
```diff
@@ -91,6 +91,15 @@ var pullImageCommand = &cli.Command{
 			EnvVars: []string{"CRICTL_PULL_TIMEOUT"},
 		},
 	},
+	Subcommands: []*cli.Command{{
+		Name:      "jsonschema",
+		Aliases:   []string{"js"},
+		Usage:     "Display the JSON schema for the pod-config.json",
```
Comment (haircommander) [inline cmd/crictl/image.go:97]: why this for the image subcommand?

Comment (haircommander) [reply cmd/crictl/image.go:97]: pedantically: seems like a bleeding of the runtime service into the image client (not like they're actually separated)

Comment (saschagrunert) [reply cmd/crictl/image.go:97]: It's added here because of `crictl pull --pod-config pod-config.[json|yam]`.

File: cmd/crictl/container.go (hunk 2 above)
Comment (SergeyKanzhelev) [reply cmd/crictl/container.go:195]: I wonder if we need to mention it at least in docs.

File: cmd/crictl/image.go (hunk 3 above)
Comment (SergeyKanzhelev) [reply cmd/crictl/image.go:97]: I am not sure I understand the logic of using it in `pull`. What is the output of a command will be?

Comment (saschagrunert) [reply cmd/crictl/image.go:97]: @SergeyKanzhelev the sandbox config provides additional context during the image pull. For example CRI-O uses it to separate pulls into dedicated cgroups: 

https://github.com/cri-o/cri-o/blob/88fe8766ede790f465e1e86b0eccba0818736b32/server/image_pull.go#L47

File: cmd/crictl/container.go (hunk 2 above)
Comment (saschagrunert) [reply cmd/crictl/container.go:195]: Yep, done

**Review by SergeyKanzhelev — APPROVED** [review]
/lgtm

**Pull request merged**